### PR TITLE
 ptr check 7214 v2

### PR DESCRIPTION
**Pull request by 1bsyl**
another  attempt for https://github.com/libsdl-org/SDL/issues/7214 

- there is a hidapi file. is this one ok? ( src/hidapi/SDL_hidapi.c )
- haven't changed "(item)" to "item"

### CODE REVIEW

**Comment by slouken** [comment]
> * there is a hidapi file. is this one ok? ( src/hidapi/SDL_hidapi.c )

Yep, that's fine.

**Comment by 1bsyl** [comment]
I have reverted the SDL_hid_api because, there was lots of include/indent changes. (maybe I run the clang-format-src.sh on it).

**Comment by 1bsyl** [comment]
I'd like not to change (item)" to "item" in the same PR, because it's already a lot of diff to handle.

full patch is here: https://patch-diff.githubusercontent.com/raw/libsdl-org/SDL/pull/8510.patch

**Comment by 1bsyl** [comment]
it also changes things like that:

```diff
-    for (devinfo = videodata->mouse_device_info; devinfo != NULL; devinfo = devinfo->next) {
+    for (devinfo = videodata->mouse_device_info; devinfo; devinfo = devinfo->next) {
```

```diff
-                    while (token != NULL) {
+                    while (token) {
```

```diff
-        SDL_assert(renderer != NULL); /* should have explicitly checked this above. */
+        SDL_assert(renderer); /* should have explicitly checked this above. */
```

**Comment by sezero** [comment]
We should expect identical compiler output before and after this, yes?
That's not what happened to me with gcc-4.4 and gcc-4.9, but it may be
because of some quirks in those old compilers.

Have you verified that binaries are identical before and after this?
(Maybe by diff'ing hexdump outputs after stripping, or by comparing
asm outputs? The patch is too large to review line by line..)

**Comment by slouken** [comment]
No, binaries won't be identical, especially in debug mode.

**Comment by sezero** [comment]
> No, binaries won't be identical, especially in debug mode.

I really meant in release mode

File: src/joystick/windows/SDL_windowsjoystick.c [hunk 1]
```diff
@@ -176,7 +176,7 @@ static DWORD CALLBACK SDL_DeviceNotificationFunc(HCMNOTIFICATION hNotify, PVOID
 static void SDL_CleanupDeviceNotificationFunc(void)
 {
     if (cfgmgr32_lib_handle) {
-        if (s_DeviceNotificationFuncHandle != NULL && CM_Unregister_Notification != NULL) {
+        if (s_DeviceNotificationFuncHandle != NULL && CM_Unregister_Notification) {
```
Comment (slouken) [inline src/joystick/windows/SDL_windowsjoystick.c:179]: Missed one here?

**Comment by slouken** [comment]
Aside from the one place I spotted, this looks good to me. @icculus, thoughts?

File: src/joystick/windows/SDL_windowsjoystick.c (hunk 1 above)
Comment (1bsyl) [reply src/joystick/windows/SDL_windowsjoystick.c:179]: the patch is 43000 lines long :)

**Comment by 1bsyl** [comment]
yes, we should check with md5sum maybe on libs...

I am trying in release, and binary size already look different .. not sure if I do correctly ?
(`cmake . -DCMAKE_BUILD_TYPE=Release` ? )

**Comment by 1bsyl** [comment]
btw "Testautomation" was ok.

**Comment by icculus** [comment]
Looks okay to me.

**Comment by slouken** [comment]
Can you do a similar patch for SDL2, so they match?

Thanks!

**Pull request merged**